deckhouse-controller/pkg/controller/module-controllers/release/controller.go

```diff
@@ -150,87 +151,82 @@ func NewModuleReleaseController(
 				if err != nil {
-					c.logger.Fatalf("Send SIGUSR2 signal failed: %s", err)
+					r.logger.Fatalf("Send SIGUSR2 signal failed: %s", err)
 				}
 			}
-			c.delayTimer.Reset(3 * time.Second)
+			r.delayTimer.Reset(3 * time.Second)
 
 		case <-ctx.Done():
 			return
 		}
 
-		c.m.Unlock()
+		r.m.Unlock()
 	}
 }
 
 // only ModuleRelease with active finalizer can get here, we have to remove the module on filesystem and remove the finalizer
-func (c *moduleReleaseReconciler) deleteReconcile(ctx context.Context, mr *v1alpha1.ModuleRelease) (ctrl.Result, error) {
-	res := ctrl.Result{}
-
+func (r *moduleReleaseReconciler) deleteReconcile(ctx context.Context, mr *v1alpha1.ModuleRelease) (res ctrl.Result, err error) {
 	// deleted release
 	// also cleanup the filesystem
-	modulePath := path.Join(c.downloadedModulesDir, mr.GetModuleName(), "v"+mr.Spec.Version.String())
+	modulePath := path.Join(r.downloadedModulesDir, mr.Spec.ModuleName, "v"+mr.Spec.Version.String())
 
-	err := os.RemoveAll(modulePath)
+	err = os.RemoveAll(modulePath)
 	if err != nil {
-		return res, err
+		return res, fmt.Errorf("remove all in %s: %w", modulePath, err)
 	}
 
 	if mr.Status.Phase == v1alpha1.PhaseDeployed {
 		extenders.DeleteConstraints(mr.GetModuleName())
-		symlinkPath := filepath.Join(c.downloadedModulesDir, "modules", fmt.Sprintf("%d-%s", mr.Spec.Weight, mr.GetModuleName()))
+		symlinkPath := filepath.Join(r.downloadedModulesDir, "modules", fmt.Sprintf("%d-%s", mr.Spec.Weight, mr.Spec.ModuleName))
 		err := os.RemoveAll(symlinkPath)
 		if err != nil {
-			return res, err
+			return ctrl.Result{}, err
 		}
 		// TODO(yalosev): we have to disable module here somehow.
 		// otherwise, hooks from file system will fail
 
 		// restart controller for completely remove module
 		// TODO: we need another solution for remove module from modulemanager
-
-		c.emitRestart("a module release was removed")
+		r.emitRestart("a module release was removed")
 	}
 
 	if !controllerutil.ContainsFinalizer(mr, fsReleaseFinalizer) {
 		return res, nil
 	}
 
 	controllerutil.RemoveFinalizer(mr, fsReleaseFinalizer)
-	err = c.client.Update(ctx, mr)
+	err = r.client.Update(ctx, mr)
 	if err != nil {
 		return res, err
 	}
 
 	return res, nil
 }
 
-func (c *moduleReleaseReconciler) createOrUpdateReconcile(ctx context.Context, mr *v1alpha1.ModuleRelease) (ctrl.Result, error) {
-	res := ctrl.Result{}
-
+func (r *moduleReleaseReconciler) createOrUpdateReconcile(ctx context.Context, mr *v1alpha1.ModuleRelease) (res ctrl.Result, err error) {
 	switch mr.Status.Phase {
 	case "":
 		mr.Status.Phase = v1alpha1.PhasePending
-		mr.Status.TransitionTime = metav1.NewTime(c.dc.GetClock().Now().UTC())
-		if err := c.client.Status().Update(ctx, mr); err != nil {
-			return res, err
+		mr.Status.TransitionTime = metav1.NewTime(r.dc.GetClock().Now().UTC())
+		if err := r.client.Status().Update(ctx, mr); err != nil {
+			return ctrl.Result{}, fmt.Errorf("update status: %w", err)
```

Comment:
if we use named return parameter, please don't change alias

Reply:
I'm not quite sure what that means in this context. Are we talking about shadowing the err variable?

Reply:
you change 
```
return res, err
```
for
```
return ctrl.Result{}, fmt.Errorf("update status: %w", err)
```

Reply:
done